jerry-core/ecma/base/ecma-gc.c

```diff
@@ -545,6 +547,8 @@ ecma_gc_free_native_pointer (ecma_property_t *property_p) /**< property */
 
     native_pointer_p = next_p;
   }
+
+  JERRY_CONTEXT (status_flags) |= ECMA_STATUS_API_AVAILABLE;
```

Comment:
These are equivalents of the `jerry_make_api_unavailable` and `jerry_make_api_available` static inline functions of jerry.c. But there, they are only called from `jerry_cleanup` and `jerry_init`. Here, they are called on every invocation of `ecma_gc_free_native_pointer`. Wont be that a performance hit?

(Note: that whole flag is strange. It is unconditionally set/cleared by `jerry_init` and `jerry_cleanup` but only checked in debug asserts. Seems like setting/clearing it should also happen in debug builds only. And then the preformance hit question would also go away.)

Reply:
I support the idea to toggle it only in debug builds. However it still does not solve the possible release build crashes, but IMO after a crash with a debug build the issue can be figured out easily.

Reply:
Note that toggling the flag in release builds would not rule out release crashes either as nobody would check the flag (because the checks are debug-only asserts).